packages/devtools_app/lib/src/memory/flutter/memory_graph_model.dart

```diff
@@ -0,0 +1,439 @@
+      final StringBuffer sb = StringBuffer();
+
+      final libraryKey = normalizeLibraryName(c.origin);
+
+      // Collect classes for each library (group by library).
+      sb.write(libraryKey);
+      final librarySbToString = sb.toString();
+      rawGroupByLibrary[librarySbToString] ??= [];
+      rawGroupByLibrary[librarySbToString].add(c);
+      sb.clear();
+
+      // Collect instances for each class (group by class)
+      for (HeapGraphElementActual instance in c.getInstances(this)) {
+        sb.write(c.name);
+        c.instancesTotalShallowSizes += instance.origin.shallowSize;
+        final classSbToString = sb.toString();
+        rawGroupByClass[classSbToString] ??= [];
+        rawGroupByClass[classSbToString].add(instance);
+        sb.clear();
+      }
+    }
+
+    return true;
+  }
+
+  bool computeFilteredGroups() {
+    // Clone groupByClass from raw group.
+    groupByClass.clear();
+    rawGroupByClass.forEach((key, value) {
+      groupByClass[key] = value.toList();
+    });
+
+    // Prune classes that are private or have zero instances.
+    groupByClass.removeWhere((className, instances) =>
+        (controller.filterZeroInstances && instances.isEmpty) ||
+        (controller.filterPrivateClasses && className.startsWith('_')) ||
+        className == internalClassName);
+
+    // Clone groupByLibrary from raw group.
+    groupByLibrary.clear();
+    rawGroupByLibrary.forEach((key, value) {
+      groupByLibrary[key] = value.toList();
+    });
+
+    // Prune libraries if all their classes are private or have zero instances.
+    groupByLibrary.removeWhere((libraryName, classes) {
+      classes.removeWhere((actual) =>
+          (controller.filterZeroInstances &&
+              actual.getInstances(this).isEmpty) ||
+          (controller.filterPrivateClasses && actual.name.startsWith('_')) ||
+          actual.name == internalClassName);
+
+      // Hide this library?
+      if (controller.libraryFilters.isLibraryFiltered(libraryName)) return true;
+
+      return controller.filterLibraryNoInstances && classes.isEmpty;
+    });
+
+    return true;
+  }
+
+  // TODO(terry): Need dominator graph for flow.
+  /// Compute all instances, needed for retained space.
+  void computeInstancesForClasses() {
+    if (!instancesComputed) {
+      for (var i = 0; i < elements.length; i++) {
+        final HeapGraphElementActual instance = elements[i];
+        instance.theClass.addInstance(instance);
+      }
+
+      instancesComputed = true;
+    }
+  }
+}
+
+abstract class HeapGraphElement {
+  /// Outbound references, i.e. this element points to elements in this list.
+  List<HeapGraphElement> _references;
+
+  void Function() referencesFiller;
+
+  List<HeapGraphElement> get references {
+    if (_references == null && referencesFiller != null) {
+      _references = [];
+      referencesFiller();
+    }
+    return _references;
+  }
+
+  String getPrettyPrint(Map<Uri, Map<String, List<String>>> prettyPrints) {
```

Comment:
avoid prefixing method names with the word `get`
this could also be a good method to document a little bit as it isn't immediately clear to me what it does.

Reply:
This was a debugging function - removed.